File: filterls/filter.rb
#!/usr/bin/env ruby

def str_to_i(str)
	num = str.to_i
  	num.to_s == str ? num : nil
end

def is_valid(line)
	segs = line.split()
	if segs.length > 1
		num = str_to_i(segs[1])
		return true if num != nil and num > 10
	end
	return false
end

ARGF.each do |line|
	puts line if is_valid(line)
end

File: filterls/filter.jl
#!/usr/bin/env julia

function filterLine(s)
	segs = split(s, " ", limit=3)
	if length(segs) > 1
		num = tryparse(Int, segs[2])
		num.hasvalue ? num.value > 10 : false
	else
		false
	end
end

for l in eachline()
	if filterLine(l)
		print("$l\n")
	end
end
